### PR TITLE
feat: `getAccessToken()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- implement `getAccessToken()`
- use `getAccessToken()` in `fetchUserInfo()`

To-do:
- LOG-1464 Fetch token by refresh token always returns id token
- add unit tests (will do in the separate PR)

## Testing

tested with `fetchUserInfo()`

- [x] when user is not authenticated, throws `.noRefreshTokenFound`
- [x] when user is authenticated and cached access token is valid, use cached token to fetch user info
- [x] when user is authenticated and cached access token is not valid (expired), use refresh token to fetch a new access token and update local tokens accordingly, then fetch user info

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Sources/Logto/Core/LogtoCore+Fetch.swift [hunk 1]
```diff
@@ -82,7 +82,7 @@ public extension LogtoCore {
 
     struct RefreshTokenTokenResponse: Codable, Equatable {
         public let accessToken: String
-        public let refreshToken: String
+        public let refreshToken: String?
```
[the remark this answers is not in the document]
Comment (demonzoo) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:85]: According to [the document](https://www.notion.so/silverhand/RefreshTokenTokenResponse-22b8a8cf606843df988141cec4e54cef), both of the tokens should be required.

image

File: Sources/Logto/Core/LogtoCore+Fetch.swift [hunk 2]
```diff
@@ -82,7 +82,7 @@ public extension LogtoCore {
...
         public let idToken: String?
         public let scope: String
```
Comment (demonzoo) [inline Sources/Logto/Core/LogtoCore+Fetch.swift:87]: This should be `scopes: String[]`, I think

File: Sources/Logto/Core/LogtoCore+Fetch.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:85]: If refresh token rotation has been turned off then it'll be optional. For now we can treat it as required.

[the remark this answers is not in the document]
Comment (demonzoo) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:85]: OK, I see. Should I help update the document to make `refreshToken` optional?

[the remark this answers is not in the document]
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:85]: idk, I think it's ok to leave it required in v1. @xiaoyijun you can make the call

File: Sources/Logto/Core/LogtoCore+Fetch.swift (hunk 2 above)
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:87]: it's the api response, just raw string according to RFC

Comment (demonzoo) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:87]: Oops, sorry, my bad

File: Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift [hunk 3]
```diff
@@ -13,13 +13,14 @@ struct ContentView: View {
     @State var isAuthenticated: Bool
     @State var authError: Error?
 
+    let resource = "https://api.logto.io"
     let client: LogtoClient?
 
     init() {
         guard let config = try? LogtoConfig(
-            endpoint: "https://logto.dev",
+            endpoint: "http://localhost:3001",
```
Comment (demonzoo) [inline Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift:21]: This is for local test purpose, right? Should revert

File: Sources/Logto/Core/LogtoCore+Fetch.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:83]: 👌

File: Sources/Logto/Core/LogtoCore+Fetch.swift [hunk 4]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 public extension LogtoCore {
     static let postHeaders: [String: String] = [
-        "Content-Type": "application/x-www-form-urlencoded",
+        "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
```
Comment (demonzoo) [inline Sources/Logto/Core/LogtoCore+Fetch.swift:12]: Is this a necessary config on swift side?

File: Sources/Logto/Core/LogtoCore+Fetch.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (demonzoo) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:83]: Let's leave it as-is then

**Review by demonzoo — APPROVED** [review]
LGTM

File: Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift (hunk 3 above)
Comment (gao-sun) [reply Demos/SwiftUI Demo/SwiftUI Demo/ContentView.swift:21]: yes

File: Sources/Logto/Core/LogtoCore+Fetch.swift (hunk 4 above)
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore+Fetch.swift:12]: no, just for testing purpose.

**Pull request merged**